sw/airborne/mcu_periph/uart.h

```diff
@@ -33,18 +33,20 @@
 #include "std.h"
 
 #ifndef UART_RX_BUFFER_SIZE
-#if defined STM32F4 || defined STM32F7 //the F4 and F7 have enough memory
-#define UART_RX_BUFFER_SIZE 256
-#else
+// Only for the STM32F1 less buffer
+#if defined(STM32F1)
```

Comment:
Here only `STM32F1` works(for the stm32 arch), since we can't include `hal.h` and else sometimes the wrong buffer size is set causing it to crash. I was doubting to remove the if statement here completely, but wasn't sure if this works for every board having a bigger buffer size. (CI should fail with boards exceeding the available RAM)